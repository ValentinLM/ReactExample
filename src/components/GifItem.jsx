import PropTyes from 'prop-types';

/* eslint-disable react/prop-types */
export const GifItem = ( { title, url } ) => {
  return (
    <div>
        <img src={ url } alt={ title } />
        <p>{ title }</p>
    </div>
  )
}

GifItem.propTypes = {
  title: PropTyes.string.isRequired,
  url: PropTyes.string.isRequired
}