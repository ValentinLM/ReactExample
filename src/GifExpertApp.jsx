import { useState } from "react";
import { AddCategory, GifGrid } from "./components";

export const GifExpertApp = () => {

  const [categories, setCategories] = useState( ['One Piece', 'Pokemon'] );

  const onAddCategory = ( newCategory ) => {
    //setCategories( categories.concat("Dragon ball"));
    if ( categories.includes( newCategory) ) return;
    setCategories( [...categories, newCategory] );
    //setCategories( cat => [...cat, 'Dragon ball']);
  }

  return (
    <>
      {/* Titulo */}
      <h1>GifExpertApp</h1>
      
      {/* Input */}
      <AddCategory onNewCategory={ (value) => onAddCategory(value) } />

      {/*Listado de Gif*/}
      {/*<button onClick={ onAddCategory }>Agregar</button>*/}
      {
        categories.map( category => ( 
          <GifGrid
            key={ category }
            category={ category }
          />
          //return <li key={ category }> { category } </li>
        ))
      }
        {/*<li> ABC </li>*/}
      {/*Item Gif*/}
      {/**/}
    </>
  )
}