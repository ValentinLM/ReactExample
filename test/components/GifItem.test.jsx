import { render, screen } from "@testing-library/react";
import { GifItem } from "../../src/components";


/* eslint-disable no-undef */
describe('Prueba de <GifItem/>', () => {
    const title = 'Harry potter';
    const url = 'https://media2.giphy.com/media/Tl2AK8HOHj7SU/giphy.gif?cid=8928e654l5ctb40tmbpx2trlv1cyz51opp3fg8ah3jgdu24p&ep=v1_gifs_search&rid=giphy.gif&ct=g'
    
    test('Debe hacer match con el snapshot', () => {
        const { container } = render( <GifItem title={title} url={url} /> );
        expect( container ).toMatchSnapshot();
    });

    test('Debe de mostrar la imagen con el URL y el ALT indicado', () => {
        render( <GifItem title={title} url={url} /> );
        //una soluciÃ³n
        //expect( screen.getByRole( 'img' ).src ).toBe( url );
        const { src, alt } = screen.getByRole( 'img' );
        expect( src ).toBe( url );
        expect( alt ).toBe( title );
    });

    test('Debe de mostrar el titulo en el componente', () => {
        render( <GifItem title={title} url={url} /> );
        expect( screen.getByText( title ) ).toBeTruthy();
    });
});