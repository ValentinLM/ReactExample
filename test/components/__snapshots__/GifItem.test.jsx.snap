// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba de <GifItem/> Debe hacer match con el snapshot 1`] = `
<div>
  <div>
    <img
      alt="Harry potter"
      src="https://media2.giphy.com/media/Tl2AK8HOHj7SU/giphy.gif?cid=8928e654l5ctb40tmbpx2trlv1cyz51opp3fg8ah3jgdu24p&ep=v1_gifs_search&rid=giphy.gif&ct=g"
    />
    <p>
      Harry potter
    </p>
  </div>
</div>
`;
